spring-cloud-gcp-samples/spring-cloud-gcp-logging-sample/src/test/java/com.example/LoggingSampleApplicationTests.java

```diff
@@ -0,0 +1,97 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.example;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.LoggingOptions;
+import com.google.cloud.logging.Payload.StringPayload;
+import com.google.common.collect.ImmutableList;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assume.assumeThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT, classes = { Application.class })
+public class LoggingSampleApplicationTests {
+
+	private static final String LOG_FILTER = "resource.type=global AND trace:%s";
+
+	@Autowired
+	private TestRestTemplate testRestTemplate;
+
+	@LocalServerPort
+	private int port;
+
+	private Logging logClient;
+
+	@BeforeClass
+	public static void checkToRun() {
+		assumeThat(
+				"Spanner Google Cloud Logging integration tests are disabled. "
+						+ "Please use '-Dit.logging=true' to enable them. ",
+				System.getProperty("it.logging"), is("true"));
+	}
+
+	@Before
+	public void setupLogging() {
+		logClient = LoggingOptions.getDefaultInstance().getService();
+	}
+
+	@Test
+	public void testLogRecordedInStackDriver() {
+		String url = String.format("http://localhost:%s/log", port);
+		String traceHeader = "gcp-logging-test-" + Instant.now().toEpochMilli();
+
+		HttpHeaders headers = new HttpHeaders();
+		headers.add("x-cloud-trace-context", traceHeader);
+		testRestTemplate.exchange(url, HttpMethod.GET, new HttpEntity<>(headers), String.class);
+
+		String logFilter = String.format(LOG_FILTER, traceHeader);
+		Page<LogEntry> logEntryPage = logClient.listLogEntries(
```

Comment:
I'm surprised that this works immediately. I would expect that you would need to wait a little bit for the trace to appear in the service.